### ISSUE TITLE
Envoy sidecar never gets ready and gives 503 with RequestAuthentication v1beta1 policy

### ISSUE THREAD

**Issue by aliabbasjaffri** [istio/istio#34826]
**Bug description**
While upgrading Istio from `v1.4.7` to `v1.6.14` on our infrastructure, I had to update old `v1alpha1` policies and resources to `v1beta1`, which i managed via the [security-policy-migration tool](https://github.com/istio-ecosystem/security-policy-migrate). After upgrading my `MeshPolicy` to respective `RequestAuthentication` and `PeerAuthentication` policies, i noticed that the `istio-proxy` sidecar fails to come up. Logs from sidecar show `503` errors:
```
2021-08-18T13:27:57.348560Z     warn    Envoy proxy is NOT ready: config not received from Pilot (is Pilot running?): cds updates: 2 successful, 0 rejected; lds updates: 0 successful, 2 rejected
2021-08-18T13:27:57.348677Z     debug   envoy client    [external/envoy/source/common/http/codec_client.cc:104] [C3] response complete
2021-08-18T13:27:57.348696Z     debug   envoy router    [external/envoy/source/common/router/router.cc:1149] [C701][S16605412471219202067] upstream headers complete: end_stream=true
2021-08-18T13:27:57.348729Z     debug   envoy http      [external/envoy/source/common/http/conn_manager_impl.cc:1649] [C701][S16605412471219202067] closing connection due to connection close header
2021-08-18T13:27:57.348743Z     debug   envoy http      [external/envoy/source/common/http/conn_manager_impl.cc:1710] [C701][S16605412471219202067] encoding headers via codec (end_stream=true):
':status', '503'
'date', 'Wed, 18 Aug 2021 13:27:56 GMT'
'content-length', '0'
'x-envoy-upstream-service-time', '1'
'server', 'envoy'
'connection', 'close'
```
Meanwhile, `istiod` was a bit more specific on the error and showed:
```
2021-08-19T12:09:31.715919Z     warn    ads     ADS:LDS: ACK ERROR sidecar~172.19.139.6~service-bridge-6c76dd4f87-hmqqz.default~default.svc.cluster.local-14 Internal:Error adding/updating listener(s) virtualInbound: Proto constraint validation failed (JwtAuthenticationValidationError.Providers[key]: ["embedded message failed validation"] | caused by JwtProviderValidationError.LocalJwks: ["embedded message failed validation"] | caused by DataSourceValidationError.InlineString: ["value length must be at least " '\x01' " bytes"]): providers {
  key: "origins-0"
  value {
    issuer: "https://keycloak.<URL>/auth/realms/<application>"
    local_jwks {
      inline_string: ""
    }
    forward: true
    from_headers {
      name: "x-amzn-oidc-accesstoken"
    }
    payload_in_metadata: "https://keycloak.<URL>/auth/realms/<application>"
  }
}
rules {
  match {
    prefix: "/"
  }
  requires {
    requires_any {
      requirements {
        provider_name: "origins-0"
      }
      requirements {
        allow_missing {
        }
      }
    }
  }
}
```
I tried to look around for a solution to this, as i thought i might be messing up the `RequestAuthentication` and/or `PeerAuthentication` policies. However, for testing purpose, i deleted my `RequestAuthentication` policy from the cluster, and all proxy sidecars were available immediately. I redeployed the `RequestAuthentication` policy and i was able to access my application as well. 

My `RequestAuthentication` policy:
```
apiVersion: security.istio.io/v1beta1
kind: RequestAuthentication
metadata:
  annotations:
    security.istio.io/alpha-policy-convert: converted from alpha authentication policy
      /default, mesh level policy
  creationTimestamp: null
  name: default
  namespace: istio-system
spec:
  jwtRules:
  - issuer: "https://keycloak.{{domain}}/auth/realms/<application>"
    jwksUri: "https://keycloak.{{domain}}/auth/realms/<application>/protocol/openid-connect/certs"
    forwardOriginalToken: true
    fromHeaders:
    - name: x-amzn-oidc-accesstoken
---
```

**Affected product area (please put an X in all that apply)**

[ ] Docs
[ ] Installation
[X] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure
[ ] Upgrade

**Affected features (please put an X in all that apply)**

[ ] Multi Cluster
[ ] Virtual Machine
[ ] Multi Control Plane

**Expected behavior**
- The envoy proxy should come up without having to delete the `RequestAuthentication` resource

**Steps to reproduce the bug**

**Version** (include the output of `istioctl version --remote` and `kubectl version --short` and `helm version --short` if you used Helm)
```
istioctl version --remote
client version: 1.6.14
control plane version: 1.6.14
data plane version: 1.6.14 (14 proxies)
```

```
kubectl version --short
Client Version: v1.22.1
Server Version: v1.18.20-eks-8c579e
WARNING: version difference between client (1.22) and server (1.18) exceeds the supported minor version skew of +/-1
```

**How was Istio installed?**
- via `istioctl install -f options.yml`
- options.yml:
```
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
spec:
  values:
    sidecarInjectorWebhook:
      rewriteAppHTTPProbe: false 
    global:
      meshExpansion:
        enabled: true
      proxy:
        logLevel: debug
    tracing:
      enabled: false
    cni:
      enabled: true
    gateways:
      istio-ingressgateway:
        type: NodePort
```

**Environment where the bug was observed (cloud vendor, OS, etc)**

Additionally, please consider running `istioctl bug-report` and attach the generated cluster-state tarball to this issue.
Refer [cluster state archive](http://istio.io/help/bugs/#generating-a-cluster-state-archive) for more details.

**Comment by aliabbasjaffri** [comment]
According to the [comment](https://github.com/istio/istio/issues/25086#issuecomment-693600793) this should have been fixed but the issue still persists in `v1.6.14`

**Comment by howardjohn** [comment]
> According to the comment this should have been fixed but the issue still persists in v1.6.14


1.6 is EOL for over a year https://istio.io/latest/news/support/announcing-1.6-eol/. To fix it, you will need to upgrade to a supported version.
